src/laboratory/templates/laboratory/sustance/list.html

```diff
@@ -18,46 +18,104 @@ <h3 class="heading-1" ><span> {% trans 'Reactive management' %}  </span></h3>
 </div>
 </div>
 
-<div class="row">
-<div class="col-md-12">
-    <div class="add-container float-end">
-        <a class="btn btn-outline-success" href="{% url 'laboratory:sustance_add' lab_pk=laboratory org_pk=org_pk %}"><i class="fa fa-plus" title="{% trans 'Add new Substance' %}"></i> {% trans 'Add' %}</a>
-    </div>
-    <br>
-  <table id="sustancetable" class="table table-striped table-bordered" style="width:100%">
-        <thead>
-            <tr>
-                <th> </th>
-                <th>{% trans 'Name' %}</th>
-                <th>{% trans 'Cas code' %}</th>
-                <th>{% trans 'Actions' %}</th>
-
-            </tr>
-        </thead>
-        <tbody>
-        </tbody>
-  </table>
-</div>
+	<div class="row">
+    <table class="table table-striped w-100 table-responsive" id="reactive_table"></table>
 </div>
+
+{% url "laboratory:api-reactive-list" org_pk lab_pk as list_obj_url %}
+	{% trans 'Create reactive' as create_obj_tittle %}
+{% include 'gentelella/blocks/modal_template.html' with form=create_form id="create_obj_modal" title=create_obj_tittle form_id="create_obj_form" url=list_obj_url %}
+
+{% url "laboratory:api-reactive-detail" org_pk lab_pk 0 as detail_obj_url %}
+	{% trans 'Update reactive' as update_obj_tittle %}
+{% include 'gentelella/blocks/modal_template.html' with form=update_form id="update_obj_modal" title=update_obj_tittle form_id="update_obj_form" url=detail_obj_url %}
+
+{% include 'gentelella/blocks/modal_template_delete.html' with form=delete_form id="delete_obj_modal" form_id="delete_obj_form" url=detail_obj_url %}
+
+
 {% endblock %}
 
 {% block js %}
 {{block.super}}
 
 <script>
+var object_urls = {
+    list_url: "{% url 'laboratory:api-reactive-list' org_pk lab_pk%}",
+    create_url: "{% url 'laboratory:api-reactive-list' org_pk lab_pk%}",
+    destroy_url: "{% url 'laboratory:api-reactive-detail' org_pk lab_pk 0 %}",
+    update_url: "{% url 'laboratory:api-reactive-detail' org_pk lab_pk 0 %}"
+}
 
-(function(){
-    $("#sustancetable").DataTable({
-        "processing": true,
-        "serverSide": true,
-        "ajax": "{% url 'laboratory:sustance_list_json' org_pk laboratory %}",
-        "language": {
-            "url": "{% get_datatables_translation %}"
-        }
-    });
-})(jQuery);
-</script>
+//laboratory:api-reactive-list
+
+datatable_inits = {
+    columns: [
+                {data: "id", name: "id", title: "ID", type: "string", visible: false},
+					        				{
+																				data: null,
+																				name: "combined_booleans",
+																				title: "Booleans",
+																				type: "string",
+																				render: function(data, type, row, meta) {
```

Comment:
Yo need to move all this render data (icons data) to serializer class, create a SerializerMethodField and return the icons data, after that call the field data here.